results/cmd/watcher/main.go

```diff
@@ -76,9 +82,11 @@ func main() {
 }
 
 type reconciler struct {
-	logger        *zap.SugaredLogger
-	client        pb.ResultsClient
-	taskRunLister listers.TaskRunLister
+	logger            *zap.SugaredLogger
+	client            pb.ResultsClient
+	taskRunLister     listers.TaskRunLister
+	pipelineclientset versioned.Interface
+	fakeserver        *server.Server
```

Comment:
The server doesn't really need to have a concept of a real vs fake client. If you look at the logic below, they are treated identically. Instead, we should make a client that conforms to the `ResultsClient` interface that's pointed at a local server instance for tests.

e.g. for tests:
```go
s := grpc.NewServer()
pb.RegisterResultsService(s, &Server{...}) // local test server

lis, err := net.Listen("tcp", "localhost:0")
if err != nil {
  ...
}
defer lis.Close()
go s.Serve(lis)
defer s.Stop()
```

Reply:
done